File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Strong.kt
package com.nouprax.markdown.core

/** Strongly emphasized (typically bold) inline content. */
public class Strong internal constructor(
    override val id: MarkupID,
    override val revision: ULong,
    override val scope: Scope,
    /** The strongly emphasized inline content in source order. */
    public val content: kotlin.collections.List<Markup>,
) : Markup {
    override fun <Result> accept(visitor: MarkupVisitor<Result>): Result = visitor.visit(this)

    override fun equals(other: Any?): Boolean = markupEquals(this, other)

    override fun hashCode(): Int = markupHashCode(this)
}

File: packages/es-markdown-core/src/model/strong.ts
import type { MarkupBase } from "./base.js";
import type { Markup } from "./markup.js";

/** Strongly emphasized (typically bold) inline content. */
export interface Strong extends MarkupBase<"strong"> {
    /** The strongly emphasized inline content in source order. */
    readonly content: readonly Markup[];
}
